cl/search/views.py

```diff
@@ -59,8 +108,18 @@ def do_search(request, rows=20, order_by=None, type=None, facet=True):
         search_form = _clean_form(request, cd, courts)
 
         if cd['type'] == 'o':
-            si = ExtraSolrInterface(settings.SOLR_OPINION_URL, mode='r')
-            results = si.query().add_extra(**build_main_query(cd, facet=facet))
+            # This is a `related:` prefix query?
+            related_prefix_match = re.search(r'^related:opinion(?P<pk>[0-9]+)(?P<fq>.*)', cd['q'])
```

Comment:
Why is it necessary that the query *start* with the `related` part? Couldn't somebody do that in the middle (or end) of a query too?

Reply:
Agree. Position within the query should not matter.